src/Marian/index.ts

```diff
@@ -292,12 +297,31 @@ export default class Marian {
       searchProperty = [searchProperty];
     }
 
+    let res: {
+      count: number;
+      facets: any[];
+    };
+    try {
+      res = await this.index.fetchFacets(query, searchProperty, filters);
+    } catch (e) {
+      console.error(
+        `Error fetching facet metadata for query ${query}, with search property ${searchProperty}, and filters ${filters}. ${JSON.stringify(
+          e
+        )} `
+      );
+      console.trace();
+      throw e;
+    }
     try {
-      const res = await this.index.fetchFacets(query, searchProperty, filters);
       res.facets = sortFacets(res.facets);
       return res;
     } catch (e) {
-      console.error(`Error fetching facet metadata: ${JSON.stringify(e)}`);
+      console.error(
```

Comment:
I'm not sure if having 2 separate try-catch blocks here is necessary since they're both only used to log and re-throw the error. In other words, we're not necessarily handling these two errors differently (nor do we need to right now).

My recommendation would be to re-consolidate the two, but keep the context on the arguments used if anything. As long as we have context for the error that occurred when fetching metadata, it should allow us to reproduce locally to debug.